pkg/security/resolvers/tags/resolver_linux.go

```diff
@@ -46,35 +55,49 @@ func (t *LinuxResolver) Start(ctx context.Context) error {
 			case <-ctx.Done():
 				return
 			case <-delayerTick.C:
-				select {
-				case workload := <-t.workloadsWithoutTags:
-					t.checkTags(workload)
-				default:
-				}
 
+			WORKLOAD:
+				for {
+					select {
+					case workload := <-t.workloadsWithoutTags:
+						t.checkTags(workload)
+					default:
+						break WORKLOAD
+					}
+				}
```

Comment:
If I understand this correctly, it may leads to an infinite loop: if we iterate on the chan and call checkTags, and if a workload can't be resolved it will be pushed back on the same chan .. and will be resolved again in the same tick?

Reply:
I mean not an infinite loop, but it defeat the purpose of capped retries